pandas/core/frame.py

```diff
@@ -1909,41 +1909,73 @@ def to_dict(self, orient: str = "dict", into=dict):
             elif orient.startswith("i"):
                 orient = "index"
 
+        object_dtype_cols = {
+            col for col, dtype in self.dtypes.items() if is_object_dtype(dtype)
+        }
+        are_all_object_dtype_cols = len(object_dtype_cols) == len(self.dtypes)
         if orient == "dict":
             return into_c((k, v.to_dict(into)) for k, v in self.items())
 
         elif orient == "list":
             return into_c(
-                (k, list(map(maybe_box_native, v.tolist()))) for k, v in self.items()
+                (
+                    k,
+                    list(map(maybe_box_native, v.tolist()))
+                    if k in object_dtype_cols
+                    else v.tolist(),
+                )
+                for k, v in self.items()
             )
 
         elif orient == "split":
+            if are_all_object_dtype_cols:
+                data = [
+                    list(map(maybe_box_native, t))
+                    for t in self.itertuples(index=False, name=None)
+                ]
+            elif object_dtype_cols:
+                # A number of ways were tried here, this solution proved to be the
+                # most optimal in general
+                data = [list(t) for t in self.itertuples(index=False, name=None)]
+                object_type_indices = [
+                    i for i, col in enumerate(self.columns) if col in object_dtype_cols
+                ]
+                for row in data:
+                    for i in object_type_indices:
+                        row[i] = maybe_box_native(row[i])
+            else:
+                data = [list(t) for t in self.itertuples(index=False, name=None)]
```

Comment:
Instead of elif here, looks like you can make it an else and then just add `if object_dtype_cols:` before L1940-1945. That way you don't have to repeat `data = ...`

Reply:
Also, is there really a speedup here in breaking out are_all_object_dtype_cols vs only some?

Reply:
@rhshadrach 

> Instead of elif here, looks like you can make it an else and then just add `if object_dtype_cols:` before L1940-1945. That way you don't have to repeat `data = ...`
I actually had it like this but then changed to what I have now as I just preferred it this way, but happy to change it to as you suggest

Reply:
I think we should prefer not having the `data = ...` line repeated as it's a (minor) violation of DRY. For the 10% penalty, I had forgotten you were looping over rows. Makes sense to keep it.

Reply:
Yes - that's correct.